src/triageTemplate.ts

```diff
@@ -86,16 +88,24 @@ export const triageBugReport = async (
   const actualBehavior = components['Actual Behavior'].raw;
   if (actualBehavior === '') missingInfo.push('Actual Behavior');
 
-  if (missingInfo.length > 0) {
+  // comment on the pr notifying user of missing required fields
+  if (!isUpdate && missingInfo.length > 0) {
     await utils.notifyMissingInfo(context, missingInfo);
+    return;
+  }
+
+  // only edit labels if this isn't the issue's initial open event
+  if (isUpdate && utils.labelExistsOnIssue(context, MISSING_INFO_LABEL)) {
```

Comment:
This control flow is more straightforward :+1:

I think we're still unintentionally reaching `utils.addIssueLabels()` on line 101 for new issues that have missing info though. Same bug in the other two `triageFoo()` methods

One way to fix this would be

```js
if (missingInfo.length > 0) {
  if (!isUpdate) {
    await utils.notifyMissingInfo(context, missingInfo);
  }
  return;
}
```

...but what I was saying earlier about making the decision once, rather than in three places, was something more like:

```js
if (missingInfo.length > 0) {
  await utils.notifyMisisngInfo(context, missingInfo, isUpdate);
  return;
}
...

async notifyMissingInfo(context, missingInfo, isUpdate) {
  if (isUpdate) {
    // do nothing; we already notified on issue creation and don't need duplicates
    return;
  }
  ...
};
```

Reply:
@codebytere re-ping :slightly_smiling_face: